### PR TITLE
Fix: API response for activating users returns 403 Forbidden

### LINKED ISSUE #81: Bug: API response for activating users returns 403 Forbidden
### Describe the bug
The backend is not able to handle the activate request and responds with:
```
{
    "detail": "Authentication credentials were not provided."
}
```

### To Reproduce
1. In `settings.py` change the email backend to console.
    `EMAIL_BACKEND = "django.core.mail.backends.console.EmailBackend"`
2. Register a new user by making a POST request on  `http://127.0.0.1:8000/api/token_auth/register/` by posting the below-mentioned data.
      ```
      {
          "username":"test",
          "password":"123",
          "confirm_password":"123",
          "email":"test@test.com"
      }
      ```
3. After this you will receive an email in the console which contains an activation link for the new account. This account verification link is in the format `http://localhost:3000/< uidb64 >/< token >`. (example: http://localhost:3000/Nw/5o6-b655d39d1a9b37aa56e2)
4. Copy these uidb64 and token and make a get request on `http://127.0.0.1:8000/api/token_auth/activate/< uidb64 >/< token >/` (For example, it will resemble like this: http://127.0.0.1:8000/api/token_auth/activate/Nw/5o6-b655d39d1a9b37aa56e2), it must return `Invalid Link` or `email-confirmed` as a response but it returns the above-mentioned error.

### Expected behavior
Invalid Link or Email confirmed must be returned as a response.
This is due to the permission_classes in `token_auth/views/register.py` which is not able to patch permission class with custom methods like activate() which uses `@api_view`.

### Screenshots
image

### Desktop (please complete the following information):
 - OS: [e.g. Ubuntu]
 - Browser [e.g. chrome]

### How to solve?
Instead of adding the `permission_classes` variable use decorators separately.
Reason: `permission_classes` doesn't patch with activate method. I also tried with `permission_classes_by_action` but it is also not working.
The tested and working method is mentioned below:
```
from rest_framework.decorators import api_view, permission_classes
from rest_framework.permissions import AllowAny

@permission_classes([AllowAny])
class RegisterView(APIView):
    
    def post(self, request, *args, **kwargs):
        pass
    
    @api_view(("GET",))
    @permission_classes([AllowAny])
    def activate(request, uidb64, token):
        pass
```

### PR DESCRIPTION
**Pull request by Amulya-coder**
### Description

Adding the decorators separately to make the activation method available without any auth credentials.

Fixes #81 

### Type of Change:

- Code

**Code/Quality Assurance Only**
- Bugfix (non-breaking change which fixes an issue)

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by Amulya-coder** [comment]
@codesankalp can you please review it

File: token_auth/views/register.py
Comment (codesankalp) [inline token_auth/views/register.py]: A blank line is needed at the end of the file @Amulya-coder

**Review by codesankalp — CHANGES_REQUESTED** [review]
QA-Checks and tests are showing as queued. Please make the suggested changes and let's see if the workflow runs again or not. @Amulya-coder

**Comment by Amulya-coder** [comment]
@codesankalp it's still failing!

**Comment by Amulya-coder** [comment]
In my terminal, it's showing some errors like `pylint(import-error)` can this may be the reason?

**Comment by codesankalp** [comment]
@Amulya-coder Follow the guidelines mentioned in README for QA-Checks and Testing.
And according to the workflows test, Black check and Flake8 check are failing.

**Comment by Amulya-coder** [comment]
@codesankalp I have run black-checks and flake-checks everything looks fine. I think failing test isn't a major problem for this issue we can ignore this.

**Comment by codesankalp** [comment]
@Amulya-coder This PR can't be merged if tests (QA-checks) do not pass.

**Comment by codesankalp** [comment]
You can use the suggestions as mentioned in the actions.
Also I can't approve this if QA-checks are failing.

**Comment by Amulya-coder** [comment]
Ok @codesankalp I will try once again.

**Comment by Amulya-coder** [comment]
@codesankalp QA checks are passed 👍

image

**Comment by Amulya-coder** [comment]
@codesankalp now can you please approve this.

**Review by codesankalp — DISMISSED** [review]
@Amulya-coder LGTM :tada: 
**Code Review**: Done 
**Testing**

**Screenshots**
image


image


image

**Comment by codesankalp** [comment]
@isabelcosta Needs your review in this PR.

**Review by devkapilbansal — DISMISSED** [review]
LGTM :+1:

**Comment by codesankalp** [comment]
I have tested it and it is working fine @devkapilbansal.
Can you test it also, so that I can add a ready to merge label?

**Comment by devkapilbansal** [comment]
Adding `Ready to Merge` since this is already been tested above.

**Comment by codesankalp** [comment]
Thanks, @Amulya-coder for your contribution. :rocket:

**Comment by isabelcosta** [comment]
@Amulya-coder @codesankalp @devkapilbansal this is failing the tests, why is that 🤔

**Comment by codesankalp** [comment]
@isabelcosta You can trigger the workflow again but the workflow will fail due to a secret-key error.

**Comment by codesankalp** [comment]
After merging of this https://github.com/anitab-org/anitab-forms-backend/pull/109
You can trigger the workflow again and it will pass.

**Comment by Amulya-coder** [comment]
@isabelcosta locally it will pass all the build tests I have checked, here it shows due to the secret key as github requires the API key to trigger the workflow which is resolved by the codesankalp in #109

**Comment by isabelcosta** [comment]
@Amulya-coder can you please update the branch? So the tests will pass?

**Comment by Amulya-coder** [comment]
ok @isabelcosta I will update it by tomorrow.

**Comment by Amulya-coder** [comment]
@isabelcosta I have updated the branch, can you please review it, sorry for the delay.

**Review by codesankalp — APPROVED** [review]

**Review by isabelcosta — APPROVED** [review]
Cool! Thank you for your contribution @Amulya-coder 🎉

**Comment by isabelcosta** [comment]
I will merge because latest changes were only to style with black formatting, and this was tested already by @codesankalp 🚀

**Pull request merged**
